File: models/Category.ts
import Realm, {BSON} from "realm";
import Transaction from "./Transaction";

class Category extends Realm.Object {
    _id!: BSON.ObjectId;
    name!: string;
    budget!: number;
    transactions!: Transaction[];

    static schema = {
      name: "Category",
      properties: {
        _id: "objectId",
        name: "string",
        budget: "int",
        transactions: "Transaction[]"
      },
      primaryKey: "_id",
    };
}

export default Category;

File: lib/models/category.dart
import 'package:realm/realm.dart';

import 'transaction.dart';

part 'category.realm.dart';

@RealmModel()
class $Category {
  @PrimaryKey()
  @MapTo('_id')
  late ObjectId id;
  
  late String name;
  late int budget;
  late List<$FinancialTransaction> transactions;
}